lib/stoplight/light.rb

```diff
@@ -3,55 +3,25 @@
 module Stoplight
   #
   # @api private use +Stoplight()+ method instead
-  class Light
-    extend Forwardable
-    include CircuitBreaker
+  class Light < CircuitBreaker
+    include Configurable
     include Lockable
     include Runnable
 
-    # @!attribute [r] data_store
-    #   @return [Stoplight::DataStore::Base]
-    def_delegator :configuration, :data_store
+    # @!attribute [r] config
+    #   @return [Stoplight::Config]
+    #   @api private
+    attr_reader :config
 
-    # @!attribute [r] threshold
-    #   @return [Integer]
-    def_delegator :configuration, :threshold
-
-    # @!attribute [r] cool_off_time
-    #   @return [Fixnum]
-    def_delegator :configuration, :cool_off_time
-
-    # @!attribute [r] window_size
-    #   @return [Float]
-    def_delegator :configuration, :window_size
-
-    # @!attribute [r] notifiers
-    #   # @return [Array<Notifier::Base>]
-    def_delegator :configuration, :notifiers
-
-    # @!attribute [r] error_notifier
-    #   # @return [Proc]
-    def_delegator :configuration, :error_notifier
-
-    # @return [String]
-    attr_reader :name
-    # @return [Stoplight::Configuration]
-    # @api private
-    attr_reader :configuration
-
-    # @param name [String]
-    # @param configuration [Stoplight::Configuration]
-    # @yield []
-    def initialize(name, configuration)
-      @configuration = configuration
-      @name = name
+    # @param config [Stoplight::Config]
+    def initialize(config)
+      @config = config
     end
 
-    private
-
-    def reconfigure(configuration)
-      @configuration = configuration
-      self
+    # @param other [any]
+    # @return [Boolean]
+    def ==(other)
```

Comment:
Where do we use this public method? 🤔

Reply:
I don't think we use this method within the Stoplight itself. The idea behind this method is to enable users to compare stoplights, for instance, during tests, and consider two lights with the same configuration to be equivalent. 

- [ ] Test `Stoplight::Light#==`